hbase-server/src/main/java/org/apache/hadoop/hbase/wal/WALSplitter.java

```diff
@@ -112,7 +114,8 @@
 
   @VisibleForTesting
   WALSplitter(final WALFactory factory, Configuration conf, Path walDir, FileSystem walFS,
-      LastSequenceId idChecker, SplitLogWorkerCoordination splitLogWorkerCoordination) {
+      LastSequenceId idChecker, SplitLogWorkerCoordination splitLogWorkerCoordination)
+      throws IOException {
```

Comment:
Throw exception on constructor is usually an anti-pattern, especially for complicated classes. You need to deal with clean up in the constructor as you can not call close from outside...